### PR TITLE
o/devicestate, daemon: introduce factory-reset mode, allow switching

### PR DESCRIPTION
**Pull request by bboozzoo**
The first bits of factory reset support.

### CODE REVIEW

**Review by stolowski — APPROVED** [review]
LGTM

File: overlord/devicestate/devicestate_systems_test.go [hunk 1]
```diff
@@ -837,17 +899,23 @@ func (s *deviceMgrSystemsSuite) TestRebootFromRecoverToRun(c *C) {
...
+		c.Assert(err, IsNil)
+
+		m, err := s.bootloader.GetBootVars("snapd_recovery_mode", "snapd_recovery_system")
+		c.Assert(err, IsNil)
+		c.Check(m, DeepEquals, map[string]string{
+			"snapd_recovery_mode":   mode,
+			"snapd_recovery_system": s.mockedSystemSeeds[0].label,
+		})
+		c.Check(s.restartRequests, DeepEquals, []restart.RestartType{restart.RestartSystemNow})
+		c.Check(s.logbuf.String(), Matches, fmt.Sprintf(`.*: rebooting into system "%s" in "%s" mode\n`, s.mockedSystemSeeds[0].label, mode))
```
Comment (anonymouse64) [inline overlord/devicestate/devicestate_systems_test.go:917]: super minor nitpick, but could use %q here

**Review by anonymouse64 — APPROVED** [review]
lgtm

**Review by mardy — APPROVED** [review]
LGTM!

**Review by pedronis — APPROVED** [review]
thanks

**Pull request merged**